atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/ShardProgressTest.java

```diff
@@ -186,6 +188,46 @@ public void repeatedlyFailingCasThrows() {
                 .isInstanceOf(CheckAndSetException.class);
     }
 
+    @Test
+    public void canResetProgressForSpecificShards() {
+        progress.updateLastSweptTimestamp(CONSERVATIVE_TEN, 8888L);
+        progress.updateLastSweptTimestamp(CONSERVATIVE_TWENTY, 8888L);
+        assertThat(progress.getLastSweptTimestamp(CONSERVATIVE_TEN)).isEqualTo(8888L);
+        assertThat(progress.getLastSweptTimestamp(CONSERVATIVE_TWENTY)).isEqualTo(8888L);
+
+        progress.resetProgressForShard(CONSERVATIVE_TEN);
+        assertThat(progress.getLastSweptTimestamp(CONSERVATIVE_TEN)).isEqualTo(RESET_TIMESTAMP);
+        assertThat(progress.getLastSweptTimestamp(CONSERVATIVE_TWENTY)).isEqualTo(8888L);
+    }
+
+    @Test
+    public void stopsTryingToResetIfSomeoneElseDid() {
+        KeyValueService mockKvs = mock(KeyValueService.class);
+        when(mockKvs.get(any(), anyMap()))
+                .thenReturn(ImmutableMap.of(DUMMY, createValue(8L)))
+                .thenReturn(ImmutableMap.of(DUMMY, createValue(4L)))
+                .thenReturn(ImmutableMap.of(DUMMY, createValue(RESET_TIMESTAMP)));
+        doThrow(new CheckAndSetException("sadness")).when(mockKvs).checkAndSet(any());
+        ShardProgress instrumentedProgress = new ShardProgress(mockKvs);
+
+        assertThatCode(() -> instrumentedProgress.resetProgressForShard(CONSERVATIVE_TEN))
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    public void repeatedlyFailingCasThrowsForReset() {
+        KeyValueService mockKvs = mock(KeyValueService.class);
+        when(mockKvs.get(any(), anyMap()))
+                .thenReturn(ImmutableMap.of(DUMMY, createValue(8L)))
+                .thenReturn(ImmutableMap.of(DUMMY, createValue(9L)))
+                .thenReturn(ImmutableMap.of(DUMMY, createValue(10L)));
+        doThrow(new CheckAndSetException("sadness")).when(mockKvs).checkAndSet(any());
+        ShardProgress instrumentedProgress = new ShardProgress(mockKvs);
+
+        assertThatCode(() -> instrumentedProgress.resetProgressForShard(CONSERVATIVE_TEN))
```

Comment:
Probably want to verify that it only throws once the value actually repeats

Reply:
Hmm, I think this is tested by `stopsTryingToResetIfSomeoneElseDid()` above - in practice you could check the CheckAndSetException's actual values but we don't get that here because of mocks, so I don't see another way of easily doing this?

Reply:
OH I meant just verify that the method was called 4 times

Reply:
added!